pettingzoo/butterfly/cooperative_pong/ball.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import pygame
+
+
+def get_small_random_value(randomizer):
+    # generates a small random value between [0, 1/100)
+    return (1 / 100) * randomizer.rand()
+
+
+class Ball(pygame.sprite.Sprite):
+    def __init__(self, randomizer, dims, speed, bounce_randomness=False):
+        self.surf = pygame.Surface(dims)
+        self.rect = self.surf.get_rect()
+        self.speed_val = speed
+        self.speed = [int(self.speed_val * np.cos(np.pi / 4)), int(self.speed_val * np.sin(np.pi / 4))]
+        self.bounce_randomness = bounce_randomness
+        self.done = False
+        self.hit = False
+        self.randomizer = randomizer
+
+    def update2(self, area, p0, p1):
```

Comment:
Is there any reason why this is named `update2` instead of `update`?

Reply:
I believe this is named `update2` instead of `update` since `Ball` inherits `pygame.sprite.Sprite` which already has an `update` function. It could probably use a more descriptive function name.